build/azure-pipelines.pr.yml

```diff
@@ -59,3 +59,27 @@ jobs:
         runInParallel: true
         vsTestVersion: latest
         runSettingsFile: '$(System.DefaultWorkingDirectory)\src\Tests\defaults.runsettings'
+    - script: |
+        cd $(Build.SourcesDirectory)
+        pip install --upgrade pip --upgrade -r $(System.DefaultWorkingDirectory)\docs\requirements.txt
+        mkdocs build --clean --config-file $(System.DefaultWorkingDirectory)\docs\mkdocs.yml
+        git fetch
+        git checkout gh-pages

```

Comment:
consider just blasting it and replacing it. might be a little less error prone?

Reply:
This seems reliable as is. I'm working on a separate PR to cleanup warnings which will also make the docs warning more obvious.